app-modules/contact/src/Filament/Resources/ContactResource/RelationManagers/ServiceRequestsRelationManager.php

```diff
@@ -69,6 +69,17 @@ public function table(Table $table): Table
     {
         return $table
             ->recordTitleAttribute('id')
+            ->modifyQueryUsing(function ($query) {
+                $query->when(! auth()->user()->hasRole('authorization.super_admin'), function (Builder $q) {
+                    return $q->whereHas('priority.type.managers', function (Builder $query): void {
+                        $query->where('teams.id', auth()->user()->teams()->first()?->getKey());
+                    })->orWhereHas('priority.type.auditors', function (Builder $query): void {
+                        $query->where('teams.id', auth()->user()->teams()->first()?->getKey());
+                    })->whereHas('respondent', function (Builder $query) {
+                        $query->where('respondent_id', $this->getOwnerRecord()->getKey());
+                    });
```

Comment:
Just curious, is this `whereHas` needed? I would assume the relation manager query already has this scoping. Or does apply the above whereHas overwrite that, making this needed?

Reply:
Yep ! It's actually overriding the method. Services requests which not related to specific contact are also showing in the list. Hence by applying custom query I prevented that issue.